src/main/java/bio/terra/service/auth/ras/EcmService.java

```diff
@@ -55,17 +60,50 @@ public PassportApi getPassportApi() {
     return new PassportApi(client);
   }
 
+  public void addRasIssuerAndType(RASv1Dot1VisaCriterion criterion) {
+    criterion.issuer(ecmConfiguration.getRasIssuer()).type(RAS_CRITERIA_TYPE);
+  }
+
   public ValidatePassportResult validatePassport(ValidatePassportRequest validatePassportRequest) {
     var passportApi = getPassportApi();
-    return passportApi.validatePassport(validatePassportRequest);
+    var result = passportApi.validatePassport(validatePassportRequest);
+
+    if (result.isValid()) {
+      var auditInfo = result.getAuditInfo();
+      var df = new SimpleDateFormat("MM-dd-yyyy HH:mm:ss z");
+      df.setTimeZone(TimeZone.getTimeZone("UTC"));
+      logger.info(
+          """
+              [Validate Passport Audit]:
+              NIH User ID: {},
+              Transaction Number: {},
+              Data Repository accessed: {},
+              Study/Data set accessed: {},
+              Date/Time of access: {}
+              """,
+          auditInfo.getOrDefault("external_user_id", "not found"),
+          auditInfo.getOrDefault("txn", "not found"),
+          "TDR",
+          ((RASv1Dot1VisaCriterion) result.getMatchedCriterion()).getPhsId(),
+          df.format(new Date(System.currentTimeMillis())));
+    }
+
+    return result;
   }
+
   /**
    * @param userReq authenticated user
    * @return the user's linked RAS passport as a JWT, or null if none exists.
    */
   public String getRasProviderPassport(AuthenticatedUserRequest userReq) {
     try {
-      return oidcApiService.getOidcApi(userReq).getProviderPassport(RAS_PROVIDER);
+      String passport = oidcApiService.getOidcApi(userReq).getProviderPassport(RAS_PROVIDER);
+      // Passports returned by OidcApi have a bug in their formatting:
+      // double quotes must be stripped if passing back to PassportApi for validation,
+      // otherwise the passport will not be considered valid JWT.
+      // This stopgap can be removed when the client is fixed:
+      // https://broadworkbench.atlassian.net/browse/ID-128
+      return passport.replace("\"", "");
```

Comment:
Ok, and I guess this won't break if they fix it and we still have this in there.  I do wonder: if there a case where there might actually be a valid quote in there?

Reply:
Good question!  A JWT is "s three Base64-URL strings separated by dots" so should not contain valid quotes ([docs](https://jwt.io/introduction/)).

Reply:
right on!  Sorry, I wasn't sure if this was the full json payload.  Presumably this is for getting rid of leading and trailing quotes?

Reply:
All good!  Yes, for leading and trailing.  I could be more specific about stripping those if you'd prefer.

Reply:
Hmm..maybe?  I wonder if there's a utility method in Apache commons that does this

Reply:
Maybe this? https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#stripAll-java.lang.String:A-java.lang.String-

Reply:
Good find, will check it out!